File: client/src/app/shared/models/team.model.ts
import { User } from "src/app/modules/users/models/user";
import { Drawing } from "..";

export interface Team {
  _id: string;

  name: string;
  description: string;

  owner: string;

  members: string[] | User[];
  memberLimit: number;

  isPrivate: boolean;
  password: string;

  drawings: string[] | Drawing[];
}

File: mobile/app/src/main/java/com/example/colorimagemobile/models/TeamModel.kt
package com.example.colorimagemobile.models

data class TeamModel(val _id: String, val name: String, val description: String, val owner: String, var members: ArrayList<String>, val drawings: ArrayList<String>, val posts: ArrayList<String>, val memberLimit: Int?)
data class CreateTeamModel(val name: String, val description: String, val memberLimit: Int?, val owner: String)